### PR TITLE
Consolidate PARAM methods

### PR DESCRIPTION
**Pull request by gpetrou**
##  Proposed changes

- Consolidate PARAM methods

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2261)

### CODE REVIEW

File: src/Common/src/Interop/Interop.PARAM.cs [hunk 1]
```diff
@@ -15,6 +15,35 @@ internal static class PARAM
         public static IntPtr FromLowHigh(int low, int high)
             => (IntPtr)((high << 16) | (low & 0xffff));
 
+        public static int ToInt(int low, int high)
```
Comment (gpetrou) [inline src/Common/src/Interop/Interop.PARAM.cs]: Not sure about the naming of these methods. Any ideas?

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs [hunk 2]
```diff
@@ -2477,7 +2477,7 @@ protected override void WndProc(ref Message m)
                 {
                     if (m.Msg == WindowMessages.WM_ACTIVATE)
                     {
-                        if (Visible && NativeMethods.Util.LOWORD(unchecked((int)(long)m.WParam)) == NativeMethods.WA_INACTIVE)
+                        if (Visible && PARAM.LOWORD(unchecked((int)(long)m.WParam)) == NativeMethods.WA_INACTIVE)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]: Shouldn't we use the overload that takes `IntPrt`?

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs [hunk 3]
```diff
@@ -932,8 +932,8 @@ protected override void WndProc(ref Message m)
                         break;
 
                     case WindowMessages.WM_NCHITTEST:
-                        Point pt = new Point((short)NativeMethods.Util.LOWORD(unchecked((int)(long)m.LParam)),
-                                             (short)NativeMethods.Util.HIWORD(unchecked((int)(long)m.LParam)));
+                        Point pt = new Point((short)PARAM.LOWORD(unchecked((int)(long)m.LParam)),
+                                             (short)PARAM.HIWORD(unchecked((int)(long)m.LParam)));
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs]: ditto

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ComponentTray.cs [hunk 4]
```diff
@@ -2556,8 +2556,8 @@ protected override void WndProc(ref Message m)
                         break;
                     case WindowMessages.WM_CONTEXTMENU:
                         // We must handle this ourselves.  Control only allows regular Windows Forms context menus, which doesn't do us much good.  Also, control's button up processing calls DefwndProc first, which causes a right mouse up to be routed as a WM_CONTEXTMENU.  If we don't respond to it here, this message will be bubbled up to our parent, which would pop up a container context menu instead of our own.
-                        int x = NativeMethods.Util.SignedLOWORD(unchecked((int)(long)m.LParam));
-                        int y = NativeMethods.Util.SignedHIWORD(unchecked((int)(long)m.LParam));
+                        int x = PARAM.SignedLOWORD(unchecked((int)(long)m.LParam));
+                        int y = PARAM.SignedHIWORD(unchecked((int)(long)m.LParam));
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ComponentTray.cs]: ditto

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ComponentTray.cs [hunk 5]
```diff
@@ -1640,7 +1640,7 @@ protected override void WndProc(ref Message m)
                     if (glyphManager != null)
                     {
                         // Get a hit test on any glyhs that we are managing this way - we know where to route appropriate  messages
-                        Point pt = new Point((short)NativeMethods.Util.LOWORD(unchecked((int)(long)m.LParam)), (short)NativeMethods.Util.HIWORD(unchecked((int)(long)m.LParam)));
+                        Point pt = new Point((short)PARAM.LOWORD(unchecked((int)(long)m.LParam)), (short)PARAM.HIWORD(unchecked((int)(long)m.LParam)));
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ComponentTray.cs]: ditto

File: src/Common/src/Interop/Interop.PARAM.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/Common/src/Interop/Interop.PARAM.cs]: They're not ideal, but ok for internal use. Ultimately I think we need to turn this into a struct and pass it around, so the names will lose any sort of ambiguity.

File: src/Common/src/Interop/Interop.PARAM.cs [hunk 6]
```diff
@@ -15,6 +15,35 @@ internal static class PARAM
...
+        public static int SignedLOWORD(IntPtr n)
+            => SignedLOWORD(unchecked((int)(long)n));
+
+        public static int SignedHIWORD(int n)
+            =>(int)(short)((n >> 16) & 0xffff);
+
+        public static int SignedLOWORD(int n)
+            => (int)(short)(n & 0xFFFF);
+
+        public static int ToLow(int n) => n & 0xffff;
```
Comment (JeremyKuhne) [inline src/Common/src/Interop/Interop.PARAM.cs]: I wouldn't do this one, use `LOWORD()`

File: src/Common/src/Interop/Interop.PARAM.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/Common/src/Interop/Interop.PARAM.cs]: For all the ones where we aren't going to/from IntPtr I'd be happier with adding a `Convert` static class. @RussKie, what are your thoughts?

Comment (RussKie) [reply src/Common/src/Interop/Interop.PARAM.cs]: Yes, we can do that.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerFrame.cs [hunk 7]
```diff
@@ -546,7 +546,7 @@ public void InvalidateOverlays(Region screenRegion)
             protected override void WndProc(ref Message m)
             {
                 base.WndProc(ref m);
-                if (m.Msg == WindowMessages.WM_PARENTNOTIFY && NativeMethods.Util.LOWORD(unchecked((int)(long)m.WParam)) == (short)WindowMessages.WM_CREATE)
+                if (m.Msg == WindowMessages.WM_PARENTNOTIFY && PARAM.LOWORD(unchecked((int)(long)m.WParam)) == (short)WindowMessages.WM_CREATE)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerFrame.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Application.ParkingWindow.cs [hunk 8]
```diff
@@ -153,8 +153,8 @@ protected override void WndProc(ref Message m)
                 base.WndProc(ref m);
                 switch (m.Msg)
                 {
-                    case Interop.WindowMessages.WM_PARENTNOTIFY:
-                        if (NativeMethods.Util.LOWORD(unchecked((int)(long)m.WParam)) == Interop.WindowMessages.WM_DESTROY)
+                    case WindowMessages.WM_PARENTNOTIFY:
+                        if (PARAM.LOWORD(unchecked((int)(long)m.WParam)) == Interop.WindowMessages.WM_DESTROY)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Application.ParkingWindow.cs]: ditto

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs (hunk 2 above)
Comment (gpetrou) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]: Done.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs (hunk 3 above)
Comment (gpetrou) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/BehaviorService.cs]: Done.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ComponentTray.cs (hunk 4 above)
Comment (gpetrou) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ComponentTray.cs]: Done.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ComponentTray.cs (hunk 5 above)
Comment (gpetrou) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ComponentTray.cs]: Done.

File: src/Common/src/Interop/Interop.PARAM.cs (hunk 6 above)
Comment (gpetrou) [reply src/Common/src/Interop/Interop.PARAM.cs]: OK.

**Pull request merged**